utils.go

```diff
@@ -10,6 +13,23 @@ func retrySleep(retries int) {
 	time.Sleep(time.Duration(retries*500) * time.Millisecond)
 }
 
+func NewRegionSpecifier(regionName string) *proto.RegionSpecifier {
+	return &proto.RegionSpecifier{
+		Type:  proto.RegionSpecifier_REGION_NAME.Enum(),
+		Value: []byte(regionName),
+	}
+}
+
+func FindKey(region *RegionInfo, key []byte) bool {
+	if region == nil {
+		return false
+	}
+	// StartKey <= key < EndKey
+	return (region.StartKey == nil || bytes.Compare(region.StartKey, key) <= 0) &&
+		(region.EndKey == nil || len(region.EndKey) == 0 ||
```

Comment:
`region.EndKey == nil` is needless, for if `region.EndKey == nil` then `len(region.EndKey) == 0`.